test/extended/oauth/oauth_ldap.go

```diff
@@ -0,0 +1,128 @@
+	)
+
+	g.It("should authenticate against an ldap server", func() {
+		g.By("setting up an OpenLDAP server")
+		ldapService, ldapCA, err := testutil.CreateLDAPTestServer(oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("deploying an oauth server")
+		caConfigMap := corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "ldapca",
+			},
+			Data: map[string]string{
+				"ca.crt": string(ldapCA),
+			},
+		}
+
+		// Configure IDP
+		ldapProvider, err := oauthutil.GetRawExtensionForOsinProvider(
+			&osinv1.LDAPPasswordIdentityProvider{
+				URL:    ldapScheme + ldapService + ldapPort + fmt.Sprintf("/%s?%s?%s", searchDN, searchAttr, searchScope),
+				BindDN: bindDN,
+				BindPassword: configv1.StringSource{StringSourceSpec: configv1.StringSourceSpec{
+					Value: bindPassword,
+				}},
+				Insecure: false,
+				CA:       caPath,
+				Attributes: osinv1.LDAPAttributeMapping{
+					ID:                []string{"dn"},
+					PreferredUsername: []string{"cn"},
+					Name:              []string{"displayName"},
+					Email:             []string{"mail"},
+				},
+			},
+		)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		// Deploy an OAuth server
+		tokenOpts, _, err := oauthutil.DeployOAuthServer(oc, []osinv1.IdentityProvider{
+			{
+				Name:            providerName,
+				UseAsChallenger: true,
+				UseAsLogin:      true,
+				MappingMethod:   "claim",
+				Provider:        *ldapProvider,
+			},
+		}, []corev1.ConfigMap{caConfigMap}, nil)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("configuring LDAP users")
+		volumeMounts, volumes := testutil.LDAPClientMounts()
+		_, errs := testutil.RunOneShotCommandPod(oc, "oneshot-ldappasswd", testutil.OpenLDAPTestImage, fmt.Sprintf("ldappasswd -x -H ldap://%s -Z -D %s -w %s -s %s cn=%s,%s -vvv", ldapService, bindDN, bindPassword, goodPass, userName, searchDN), volumeMounts, volumes, nil, 5*time.Minute)
+		o.Expect(errs).To(o.BeEmpty())
+
+		g.By("ensuring that you cannot authenticate with a bad password")
+		_, err = oauthutil.RequestTokenForUser(tokenOpts, userName, badPass)
+		o.Expect(err).To(o.HaveOccurred())
+
+		g.By("authenticating with LDAP user")
+		person1Token, err := oauthutil.RequestTokenForUser(tokenOpts, userName, goodPass)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		// Make sure we can use the token, and it represents who we expect
+		userConfig := restclient.Config{}
+		userConfig.Host = oc.AdminConfig().Host
+		userConfig.CAFile = oc.AdminConfig().CAFile
+		userConfig.CAData = oc.AdminConfig().CAData
+		userConfig.BearerToken = person1Token
+
+		// Confirm user name
+		user, err := userv1client.NewForConfigOrDie(&userConfig).Users().Get("~", metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(user.Name).Should(o.BeEquivalentTo(userName))
+
+		// Clean up mapped identity and user.
+		defer userv1client.NewForConfigOrDie(oc.AdminConfig()).Identities().Delete(fmt.Sprintf("%s:%s", providerName, myUserDNBase64), nil)
```

Comment:
nit: store `userv1client.NewForConfigOrDie(oc.AdminConfig())` as a var to avoid copy pasta.